src/test/java/de/tum/in/www1/artemis/service/messaging/services/LectureServiceConsumerTest.java

```diff
@@ -0,0 +1,166 @@
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        this.message = mock(Message.class);
+        setCaptor = ArgumentCaptor.forClass(Set.class);
+
+        SecurityUtils.setAuthorizationObject();
+        this.database.addUsers(10, 10, 0, 10);
+        List<Course> courses = this.database.createCoursesWithExercisesAndLectures(true);
+        Course course = this.courseRepository.findByIdWithExercisesAndLecturesElseThrow(courses.get(0).getId());
+        Lecture lecture = course.getLectures().stream().findFirst().get();
+        TextExercise textExercise = textExerciseRepository.findByCourseIdWithCategories(course.getId()).stream().findFirst().get();
+        // Add user that is not in the course
+        userRepository.save(ModelFactory.generateActivatedUser("student42"));
+
+        // Setting up a lecture with various kinds of content
+        ExerciseUnit exerciseUnit = database.createExerciseUnit(textExercise);
+        AttachmentUnit attachmentUnit = database.createAttachmentUnit();
+        VideoUnit videoUnit = database.createVideoUnit();
+        TextUnit textUnit = database.createTextUnit();
+
+        lecture = database.addLectureUnitsToLecture(lecture, Set.of(exerciseUnit, attachmentUnit, videoUnit, textUnit));
+        this.user1 = userRepository.findUserWithGroupsAndAuthoritiesByLogin("student1").get();
+        this.user2 = userRepository.findUserWithGroupsAndAuthoritiesByLogin("student42").get();
+        this.exercises = lecture.getLectureUnits().stream().filter(unit -> unit instanceof ExerciseUnit).map(unit -> ((ExerciseUnit) unit).getExercise())
+                .collect(Collectors.toSet());
+    }
+
+    @AfterEach
+    public void resetDatabase() {
+        database.resetDatabase();
+    }
+
+    @Test
+    public void testGetExercisesAndRespond() throws JMSException {
+        UserExerciseDTO dto = new UserExerciseDTO(exercises, user1);
+        when(message.getBody(UserExerciseDTO.class)).thenReturn(dto);
+        when(message.getJMSCorrelationID()).thenReturn(Integer.toString(dto.hashCode()));
+
+        lectureServiceConsumer.filterExercisesAndRespond(message);
+
+        verify(jmsTemplate, atLeastOnce()).convertAndSend(eq(MessageBrokerConstants.LECTURE_QUEUE_GET_EXERCISES_RESPONSE), setCaptor.capture(), any());
+        assertThat(setCaptor.getValue()).isEqualTo(exercises);
+    }
+
+    @Test
+    public void testGetExercisesAndRespond_shouldResponseWithEmptyList() throws JMSException {
+        UserExerciseDTO dto = new UserExerciseDTO(exercises, user2);
+        when(message.getBody(UserExerciseDTO.class)).thenReturn(dto);
+        when(message.getJMSCorrelationID()).thenReturn(Integer.toString(dto.hashCode()));
+
+        lectureServiceConsumer.filterExercisesAndRespond(message);
+
+        verify(jmsTemplate, atLeastOnce()).convertAndSend(eq(MessageBrokerConstants.LECTURE_QUEUE_GET_EXERCISES_RESPONSE), setCaptor.capture(), any());
+        assertThat(setCaptor.getValue().size()).isEqualTo(0);
```

Comment:
```suggestion
        assertThat(setCaptor.getValue()).isEmpty();
```
multiple occurrences in multiple changed files